linera-base/src/identifiers.rs

```diff
@@ -453,6 +497,56 @@ impl StreamId {
     }
 }
 
+/// The result of an events from index.
+#[derive(
+    Debug,
+    Eq,
+    PartialEq,
+    Ord,
+    PartialOrd,
+    Clone,
+    Hash,
+    Serialize,
+    Deserialize,
+    WitLoad,
+    WitStore,
+    WitType,
+    SimpleObject,
+)]
+pub struct IndexAndEvent {
+    /// The index of the found index.
+    pub index: u32,
+    /// The event being returned.
+    pub event: Vec<u8>,
+}
+
+impl fmt::Display for StreamId {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        Display::fmt(&self.application_id, f)?;
+        Display::fmt(":", f)?;
+        Display::fmt(&self.stream_name, f)
+    }
+}
+
+impl std::str::FromStr for StreamId {
+    type Err = anyhow::Error;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        let parts = s.split(':').collect::<Vec<_>>();
```

Comment:
I see; I guess if we use the colon in the generic application ID as well, we have to use `rsplit_once` here.

Reply:
Ok, done. Thanks.